csclient/csclient.go

```diff
@@ -37,6 +37,7 @@ var ServerURL = "https://api.jujucharms.com/charmstore"
 // Client represents the client side of a charm store.
 type Client struct {
```

Comment:
I'm not sure why we feel we have to hide the parameters here, I would have thought that something more like http.Client where you can just set the parts as necessary would be better. Don't change anything just a passing thought.